doc/api/esm.md

```diff
@@ -897,6 +940,7 @@ success!
 
 [CommonJS]: modules.html
 [ECMAScript-modules implementation]: https://github.com/nodejs/modules/blob/master/doc/plan-for-new-modules-implementation.md
+[package exports]: #esm_package_exports
```

Comment:
We try to keep these bottom references in ASCII order. One day, we'll get a lint rule for that. In the meantime, can  you move this one to between `dynamic instantiate hook` and `special scheme` below?